app/models/spree/payment_method_decorator.rb

```diff
@@ -2,12 +2,12 @@
   has_many :store_payment_methods
   has_many :stores, :through => :store_payment_methods
 
-  def self.available(display_on = 'both', store = nil)
-    result = all.select do |p|
-      p.active &&
-        (p.environment == Rails.env || p.environment.blank?) &&
-        (store.nil? || store.payment_methods.empty? || store.payment_methods.include?(p)) &&
-        (p.display_on == display_on.to_s || p.display_on.blank?)
+  class << self
+    alias_method :available_without_store, :available
+    def available(display_on = 'both', store = nil)
+      available_without_store(display_on).select do |p|
+        store.nil? || store.payment_methods.empty? || store.payment_methods.include?(p)
```

Comment:
this isn't your code, but i find it a bit odd that we would use another store's payment method if this store doesn't have any. This change is definitely an improvement, though it would be nice to not override the method and provide some sort of 'available_filter' that can be added to this class.